### PR TITLE
throw on unexpected keys in `expectBadge()`

### LINKED ISSUE #7276: `expectBadge()` should throw on unexpected keys
:clipboard: **Description**

This, for example, should fail:

https://github.com/badges/shields/blob/be013e175b0048f796817c431773d5a069a9a31d/services/spiget/spiget-download-size.tester.js#L10

### PR DESCRIPTION
**Pull request by chris48s**
Closes #7276

### CODE REVIEW

File: services/docsrs/docsrs.tester.js [hunk 1]
```diff
@@ -14,5 +14,5 @@ t.create('Getting latest version works')
   .get('/rand/latest.json')
   .expectBadge({
     label: 'docs',
-    messsage: Joi.allow('passing', 'failing'),
+    message: Joi.allow('passing', 'failing'),
```
Comment (chris48s) [inline services/docsrs/docsrs.tester.js:17]: :smile:

**Comment by chris48s** [comment]
Test failures are unrelated but we need to do another triage of consistently failing tests

File: services/docsrs/docsrs.tester.js (hunk 1 above)
Comment (calebcartwright) [reply services/docsrs/docsrs.tester.js:17]: Ha. Spelling common words is surprisingly difficult sometimes

**Review by calebcartwright — APPROVED** [review]
Nice :+1:

File: services/docsrs/docsrs.tester.js (hunk 1 above)
Comment (chris48s) [reply services/docsrs/docsrs.tester.js:17]: tbh I was slightly surprised this only kicked up the one

**Pull request merged**